resources/share.rb

```diff
@@ -243,7 +243,7 @@ def update_permissions
       # set permissions for a brand new share OR
       # update permissions if the current state and desired state differ
       next unless permissions_need_update?(perm_type)
-      grant_command = "Grant-SmbShareAccess -Name '#{new_resource.share_name}' -AccountName \"#{new_resource.send("#{perm_type}_users").join('","')}\" -Force -AccessRight #{perm_type}"
+      grant_command = "Grant-SmbShareAccess -Name '#{new_resource.share_name}' -AccountName #{new_resource.send("#{perm_type}_users").join('","')} -Force -AccessRight #{perm_type}"
```

Comment:
The problem we have now is we get a value of 
`Grant-SmbShareAccess -Name 'read_only' -AccountName BUILTIN\Users","foo","bar`

Based on microsoft documentation we need it in this format:
> Specifies the name of the trustee. Use a comma-separated list to specify multiple trustees.
`Grant-SmbShareAccess -Name 'read_only' -AccountName "BUILTIN\Users","foo","bar"`

Is this not what the original code was doing?

Reply:
Sorry for my inattention, it's a bug with my array which i passed to mentioned resource.